docs/data-sources/dms_kafka_flavors.md

```diff
@@ -0,0 +1,143 @@
+---
+subcategory: "Distributed Message Service (DMS)"
+---
+
+# flexibleengine_dms_kafka_flavors
+
+Use this data source to get the list of available flavor details within FlexibleEngine.
+
+## Example Usage
+
+### Query the list of kafka flavors for cluster type
+
+```hcl
+data "flexibleengine_dms_kafka_flavors" "test" {
+  type = "cluster"
+}
+```
+
+### Query the kafka flavor details of the specified ID
+
+```hcl
+data "flexibleengine_dms_kafka_flavors" "test" {
+  flavor_id = "c6.2u4g.cluster"
+}
+```
+
+### Query list of kafka flavors that available in the availability zone list
+
+```hcl
+variable "az1" {}
+variable "az2" {}
+
+data "flexibleengine_dms_kafka_flavors" "test" {
+  availability_zones = [
+    var.az1,
+    var.az2,
+  ]
+}
+```
+
+## Argument Reference
+
+* `region` - (Optional, String) Specifies the region in which to obtain the dms kafka flavors.
+  If omitted, the provider-level region will be used.
+
+* `flavor_id` - (Optional, String) Specifies the DMS flvaor ID, e.g. **c6.2u4g.cluster**.
+
+* `storage_spec_code` - (Optional, String) Specifies the disk IO encoding.
+  + **dms.physical.storage.high.v2**: Type of the disk that uses high I/O.
+  + **dms.physical.storage.ultra.v2**: Type of the disk that uses ultra-high I/O.
+
+* `type` - (Optional, String) Specifies flavor type. The valid values are **single** and **cluster**.
+
+* `arch_type` - (Optional, String) Specifies the type of CPU architecture, e.g. **X86**.
+
+* `availability_zones` - (Optional, List) Specifies the list of availability zones with available resources.
+
+* `charging_mode` - (Optional, String) Specifies the flavor billing mode.
+  The valid valus are **prePaid** and **postPaid**.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The data source ID.
+
+* `versions` - The supported flavor versions.
+
+* `flavors` - The list of flavor details.
```

Comment:
Usually, the field name, structure name and block name should keep consistent
there are some fields such as `ios`,`support_features`,`properties` have the same problem in this document 
```markdown
* `flavors` - The list of flavor details.
The [flavors](#dms_kafka_flavors) structure is documented below.
```